src/status.c

```diff
@@ -0,0 +1,28 @@
+#include "status.h"
+
+#include <stdbool.h>
+#include <stdlib.h>
+
+static Status global_status = {.code = STATUS_CODE_OK, .caller = NULL, .message = "" };
```

Comment:
Since it's static, it's automatically zeroed on boot. You can use a designated initializer to set the message to an empty string, which I assume is to avoid having to check if the message is empty before printing.